### PR TITLE
Add tests for the IdP picker

### PR DESCRIPTION
**Pull request by richvdh**
This adds some testing to the Identity-Provider picker which was added in #9017. The tests aren't amazing, but hopefully they will suffice to let us know if something completely breaks.

There are a couple of preparatory commits to clean up some stuff in the test file, which should be reviewable on their own.

### CODE REVIEW

File: synapse/rest/client/v1/login.py [hunk 1]
```diff
@@ -319,9 +319,9 @@ def __init__(self, hs: "HomeServer"):
         # register themselves with the main SSOHandler.
         if hs.config.cas_enabled:
             hs.get_cas_handler()
-        elif hs.config.saml2_enabled:
+        if hs.config.saml2_enabled:
             hs.get_saml_handler()
-        elif hs.config.oidc_enabled:
+        if hs.config.oidc_enabled:
```
Comment (erikjohnston) [inline synapse/rest/client/v1/login.py:324]: Is this not a fixing something real?

**Review by erikjohnston — APPROVED** [review]

File: synapse/rest/client/v1/login.py (hunk 1 above)
Comment (richvdh) [reply synapse/rest/client/v1/login.py:324]: ah, not really. In a real synapse, the OIDC and SAML handlers get instantiated via other codepaths, so it only bit in the test environment.

Comment (erikjohnston) [reply synapse/rest/client/v1/login.py:324]: Ah, fair

Comment (richvdh) [reply synapse/rest/client/v1/login.py:324]: (this code is just belt-and-braces to make sure that the handlers definitely get instantiated if we are on a worker that hosts this endpoint.)

**Pull request merged**